cmd/zdb/zdb.c

```diff
@@ -2360,6 +2360,42 @@ dump_nvlist_stats(nvlist_t *nvl, size_t cap)
 	nvlist_free(stats.zns_boolean);
 }
 
+static void
+dump_unique_config(nvlist_t *config, char *buf, size_t buflen, int l,
+    zio_cksum_t cksum[VDEV_LABELS])
+{
+	size_t config_size;
+	int prev_label;
+	int duplicate_config;
+
+	if (nvlist_size(config, &config_size, NV_ENCODE_XDR) < 0)
+		config_size = buflen;
+
```

Comment:
`nvlist_size()` returns positive errno values. As coded won't this will leave `config_size` uninitialized on errors?